src/iconv.jl

```diff
@@ -1,9 +1,61 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
 module iconv
-import Base: close, eof, flush, read, readall, write
+import Base: close, eof, flush, read, readall, write, show
 import Base.Libc: errno, strerror, E2BIG, EINVAL, EILSEQ
 export StringEncoder, StringDecoder, encode, decode
+export StringEncodingError, OutputBufferError, IConvError
+export InvalidEncodingError, InvalidSequenceError, IncompleteSequenceError
+
+
+abstract StringEncodingError
+
+# specified encodings are not supported
+type InvalidEncodingError <: StringEncodingError
+    msg::UTF8String
```

Comment:
You could still define the method to return the `msg` field if it's actually there for the considered type. So I don't think we'd lose in generality at all.